Sources/Basics/Triple+Basics.swift

```diff
@@ -64,13 +64,11 @@ extension Triple {
     /// This is currently meant for Apple platforms only.
     public func tripleString(forPlatformVersion version: String) -> String {
         precondition(isDarwin())
-        // This function did not handle triples with a specific environments and
-        // object formats previously to using SwiftDriver.Triple and still does
-        // not.
         return """
             \(self.archName)-\
             \(self.vendorName)-\
-            \(self.osNameUnversioned)\(version)
+            \(self.osNameUnversioned)\(version)\
+            \(self.environmentName.isEmpty ? "" : "-\(self.environmentName)")
```

Comment:
@rauhul could you please confirm whether this is correct? Not sure if I'm missing anything else here.

Reply:
I'm currently on vacation and not critically thinking, however until im back (aug 28th) @artemcm may be able to provide some feedback.